app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -76,16 +82,42 @@ class MainActivity : AppCompatActivity() {
         unregisterReceiver(broadcastReceiver)
     }
 
-    fun enableButton() {
+    private fun enableButton() {
         binding.buttonGetWeather.setOnClickListener {
-            if(binding.textViewLocalization.text.isEmpty())
-                binding.progressBar.visibility = View.VISIBLE
-            Intent(this, LocalizationService::class.java).also { intent ->
-                bindService(intent, connection, Context.BIND_AUTO_CREATE)
+            getLocalization()
+            if (!NotificationManagerCompat.from(this).areNotificationsEnabled()) {
```

Comment:
When user is asked to enable notifications? After click on button I get only toast "Enable notifications to get the hourly weather.", weather is not refreshed

Reply:
I was having Internet issues, refreshing weather works fine. Notifications still has to be manually checked in settings

Reply:
I'll work on setting notification on when starting the app.